### PR TITLE
Bump(requirements): Support newer jsonschema versions

### PR DESCRIPTION
**Pull request by ClausHolbechArista**
## Change Summary

Remove limit on jsonschema version

## Component(s) name

Schema tooling

## Proposed changes

- Clean up leftovers of support for "$ref" in the schema tooling.
  - This was removed from the validator itself, but was kept on the AvdSchema class.
  - Schema will be fully resolved before starting validation, so no need for it.
- Refactor the AvdSchemaResolver to use the "referencing" library instead of the old "jsonschema<4.18".
  - Completely rewritten, so is no longer dependent on `jsonschema`.
  - This adds a new requirement of `referencing>=0.35.0`.
  - Since the "referencing" library is very picky about the `$ref` format (fails badly), all refs have been adjusted to follow the supported scheme, and the `avd_meta_schema` has been updated with a `pattern` on the `$ref` field.
- During testing we found out that our minimum requirements are not compatible, so we are also adding a test case for all minimum requirements and bumped a few of them (mdtoc) to get everything in order. In practice noone should hit any obstacles with this change, since `pip` was never able to resolve a working combination with the older versions.

## How to test

- Tested molecule with older and newer jsonschema
- Tested PyAVD with older and newer jsonschema

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Review by carlbuchmann — APPROVED** [review]
LGTM

File: .github/workflows/pull-request-management.yml
Comment (gmuloc) [inline .github/workflows/pull-request-management.yml:216]: do we need this line - is not simply using the default one?

File: ansible_collections/arista/avd/docs/pyavd.md
Comment (gmuloc) [inline ansible_collections/arista/avd/docs/pyavd.md]: does not look align with release notes and 4.10.3

File: ansible_collections/arista/avd/plugins/plugin_utils/schema/avdschemaresolver.py
Comment (gmuloc) [inline ansible_collections/arista/avd/plugins/plugin_utils/schema/avdschemaresolver.py:85]: if it is a static method should we not be using `<CLASS>.create_registry`? It works with self from what I can read so no real opinion here - just questioning how we want to do this

File: ansible_collections/arista/avd/plugins/plugin_utils/schema/avdschemaresolver.py
Comment (gmuloc) [inline ansible_collections/arista/avd/plugins/plugin_utils/schema/avdschemaresolver.py:89]: docstring

File: ansible_collections/arista/avd/plugins/plugin_utils/schema/avdschemaresolver.py
Comment (gmuloc) [inline ansible_collections/arista/avd/plugins/plugin_utils/schema/avdschemaresolver.py:118]: should we look at changing this method later?

File: ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py
Comment (gmuloc) [inline ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py:131]: I am not sure while reading the code where the error handling went. Are we still raising on resolving errors?

File: ansible_collections/arista/avd/requirements-minimum.txt
Comment (gmuloc) [inline ansible_collections/arista/avd/requirements-minimum.txt:1]: would probably be good to be able to build this programmatically from out requirement file

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/defs_network_services.schema.yml
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/defs_network_services.schema.yml:714]: Are all these modifications forced by the upgrade or is it more for consistency purposes and something we should make reviewer aware of?
asking as in some places the PR removes the leading `/` after `#` but not everywhere

File: python-avd/tox.ini
Comment (gmuloc) [inline python-avd/tox.ini:23]: so adding referencing but not json_schema?

**Review by gmuloc — COMMENTED** [review]
Thanks for tackling this!

Generic question
* Can we add an explanation in the PR description to explain why we are bumping md_toc? It is not clear to me reading the code.

File: ansible_collections/arista/avd/plugins/plugin_utils/schema/avdschemaresolver.py
Comment (gmuloc) [reply ansible_collections/arista/avd/plugins/plugin_utils/schema/avdschemaresolver.py:118]: reusing things from DRAFT7 as per referencing library

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/defs_network_services.schema.yml
Comment (gmuloc) [reply ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/defs_network_services.schema.yml:714]: there is a pattern - the `/` can be omitted if there is nothing after `#`, otherwise it is required.

File: python-avd/tox.ini
Comment (gmuloc) [reply python-avd/tox.ini:23]: these are extra dependency and depending on the version of json-schema this is needed to add referencing.

File: ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py
Comment (gmuloc) [reply ansible_collections/arista/avd/plugins/plugin_utils/schema/store.py:131]: no change in behavior and this is handled inside resolve

File: ansible_collections/arista/avd/docs/pyavd.md
Comment (carlbuchmann) [reply ansible_collections/arista/avd/docs/pyavd.md]: fixed in 66384c3

**Comment by ClausHolbechArista** [comment]
> * Can we add an explanation in the PR description to explain why we are bumping md_toc? It is not clear to me reading the code.

@gmuloc done.

**Review by gmuloc — APPROVED** [review]

File: .github/workflows/pull-request-management.yml
Comment (gmuloc) [inline .github/workflows/pull-request-management.yml:213]: Hmm so it looks like it was needed .. and not picking up the default vairable .. my bad..

**Review by gmuloc — CHANGES_REQUESTED** [review]
requesting changes to block merge

**Review by gmuloc — APPROVED** [review]
LGTM once CI passes

**Pull request merged**
